### DISCUSSION TITLE
[Feature v4] External proxy support

### DISCUSSION THREAD

**Discussion by MrSquaare** [coollabsio/coolify#3132] [Feature Requests]
The ability to use a custom external proxy

Currently, in version 4-beta.23, it is possible to choose a custom destination, which can only work for an external Traefik that:
- Use host network
- Have entrypoints named http and https
- Have a certresolver named letsencrypt
- Have a middleware http-to-https

Otherwise, it won't work.

Solutions:
- Allow the user to define one or more docker compose files and then merge them together
  - Via a custom merge
  - Via compose specification: https://docs.docker.com/compose/multiple-compose-files/merge/
    - This solution requires that the service has always the same name (not suffixed by the build timestamp)

**Comment by EAliakbar** [comment]
I created a mapper to use coolify generated Traefik configs with an external Traefik instance, it's not on a par with features we used to have in V3 with[ traefik-http-provider-mapper](https://github.com/beaussan/traefik-http-provider-mapper) but it supports changing entrypoints and certresolver names at the moment
I've tested it with coolify panel itself and it works
Let me know what you think about it
https://github.com/EAliakbar/traefik-file-provider-mapper

**Comment by MrSquaare** [comment]
> I created a mapper to use coolify generated Traefik configs with an external Traefik instance, it's not on a par with features we used to have in V3 with[ traefik-http-provider-mapper](https://github.com/beaussan/traefik-http-provider-mapper) but it supports changing entrypoints and certresolver names at the moment I've tested it with coolify panel itself and it works Let me know what you think about it [EAliakbar/traefik-file-provider-mapper](https://github.com/EAliakbar/traefik-file-provider-mapper)

Didn't see the comment, cool alternative to traefik-http-provider-mapper!

External proxy is partially supported for proxies who support Docker labels via custom labels feature (for now, only for applications)

PS: I think this issue should be converted to a discussion since, if I understand, issues will be only for bugs

**Comment by swissbyte** [comment]
Is this related to #1483 ?

**Comment by peaklabs-dev** [comment]
Generally support external proxy. If none is selected, there may be a proxy on the server that manages things.
